data_enrichment.ipynb

```diff
@@ -0,0 +1,505 @@
+       "      <td></td>\n",
+       "      <td></td>\n",
+       "      <td></td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>Home Depot</td>\n",
+       "      <td></td>\n",
+       "      <td></td>\n",
+       "      <td></td>\n",
+       "      <td></td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>LinkedIn Corporation</td>\n",
+       "      <td></td>\n",
+       "      <td></td>\n",
+       "      <td></td>\n",
+       "      <td></td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>General Dynamics</td>\n",
+       "      <td></td>\n",
+       "      <td></td>\n",
+       "      <td></td>\n",
+       "      <td></td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "                company number of employees NAICS code(s)  \\\n",
+       "0                 Apple                                     \n",
+       "1         Canadian Tire                                     \n",
+       "2            Home Depot                                     \n",
+       "3  LinkedIn Corporation                                     \n",
+       "4      General Dynamics                                     \n",
+       "\n",
+       "  headquarter location founded year  \n",
+       "0                                    \n",
+       "1                                    \n",
+       "2                                    \n",
+       "3                                    \n",
+       "4                                    "
+      ]
+     },
+     "execution_count": 208,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "data = {\n",
+    "    'company': [\"Apple\", 'Canadian Tire', 'Home Depot', 'LinkedIn Corporation', 'General Dynamics'],\n",
+    "    'number of employees': ['', '', '', '', ''],\n",
+    "    'NAICS code(s)': ['', '', '', '', ''],\n",
+    "    'headquarter location': ['', '', '', '', ''],\n",
+    "    'founded year': ['', '', '', '', ''],\n",
+    "}\n",
+    "\n",
+    "# Convert to DataFrame\n",
+    "df = pd.DataFrame(data)\n",
+    "\n",
+    "# Display the DataFrame\n",
+    "df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Searching and getting results with `You.com` API\n",
+    "\n",
+    "We will obtain the missing values with a simple call to `You.com` API"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 209,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "import os\n",
+    "\n",
+    "# obtain the API key from the environment\n",
+    "headers = {'x-api-key': os.environ['YDC_API_KEY']}\n",
+    "\n",
+    "def get_research_data(company_name, missing_cols, mode=\"research\"):\n",
```

Comment:
@christeefy, what's the latest recommended way to do this? Can they import a retriever instead here?

Reply:
Retriever implementations are framework-specific. They can import a retriever — the question becomes more about whether we want this example to be framework-agnostic or not. If yes to being framework-agnostic, then pinging the API directly would be the approach.

Reply:
Oh my bad, this is the chat API, which is implemented as an LLM (instead of a retriever). It is still framework-specific, so my comments above still apply.

Reply:
Do you know which retrievers have been merged already?
We have examples for langchain and llama index. Any others that they could use here to show off what we have done? If not, I'd suggest going with the llama-index one as we only have one example for that one right now.

Reply:
@rodrigo-georgian is llama-index using research API?

Reply:
@rodrigo-georgian All framework's You retrievers support the search API. Langchain's support News as well. 

PRs have been put up across all frameworks to update all their implementations to support News but not yet merged by the authors.

Reply:
PRs for the chat API have not been created until YDC is ready.

Reply:
Discussed with @rodrigo-georgian offline. This example showcases research API. We will leave the implementation as is for now to use rest api and switch with langchain/dspy framework once available.